RadixWallet/Clients/OnLedgerEntitiesClient/Helpers/OnLedgerEntitiesClient+CreateEntity.swift

```diff
@@ -349,10 +349,14 @@ extension OnLedgerEntitiesClient {
 			account.atLedgerState,
 			refresh
 		).compactMap(\.resourcePool)
-		let allResourceAddresses = pools.flatMap { pool in
-			[pool.poolUnitResourceAddress] +
-				pool.resources.nonXrdResources.map(\.resourceAddress) +
-				(pool.resources.xrdResource.map { [$0.resourceAddress] } ?? [])
+
+		var allResourceAddresses: [ResourceAddress] = []
+		for pool in pools {
+			allResourceAddresses.append(pool.poolUnitResourceAddress)
+			allResourceAddresses += pool.resources.nonXrdResources.map(\.resourceAddress)
```

Comment:
I think not using `+` will lead to better formance. Can you use `append:contentsOf` instead?